src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorLSPTextViewConnectionListener.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor
+{
+    [ContentType(RazorLSPContentTypeDefinition.Name)]
+    [TextViewRole(PredefinedTextViewRoles.Document)]
+    [Export(typeof(ITextViewConnectionListener))]
+    internal class RazorLSPTextViewConnectionListener : ITextViewConnectionListener
+    {
+        private readonly LSPDocumentManagerBase _lspDocumentManager;
+
+        [ImportingConstructor]
+        public RazorLSPTextViewConnectionListener(
+            LSPDocumentManager lspDocumentManager)
+        {
+            if (lspDocumentManager is null)
+            {
+                throw new ArgumentNullException(nameof(lspDocumentManager));
+            }
+
+            _lspDocumentManager = lspDocumentManager as LSPDocumentManagerBase;
+
+            Debug.Assert(_lspDocumentManager != null);
```

Comment:
>Debug.Assert(_lspDocumentManager != null); [](start = 12, length = 42)

Would it make more sense to throw here rather than null ref later if this is null?

Reply:
Ya you're definitely right. Should throw.

Reply:
Seems like the wrong exception type.

---
In reply to: [381459432](https://github.com/dotnet/aspnetcore-tooling/pull/1602#discussion_r381459432) [](ancestors = 381459432)